File: backend/app/schemas/notification.py
"""Pydantic schemas for the notifications feature."""

from datetime import datetime
from uuid import UUID

from app.models.enums import NotificationType
from app.schemas.common import ORMBaseSchema


class NotificationReadSchema(ORMBaseSchema):
    id: UUID
    user_id: UUID
    type: NotificationType
    title: str
    message: str
    is_read: bool
    related_entity_type: str | None
    related_entity_id: UUID | None
    created_at: datetime

File: frontend/src/types/notification.ts
/** Mirrors the backend's `app.schemas.notification` contracts. */

export type NotificationType =
  | "appointment_reminder"
  | "medication_reminder"
  | "report_upload_success"
  | "general";

export interface Notification {
  id: string;
  user_id: string;
  type: NotificationType;
  title: string;
  message: string;
  is_read: boolean;
  related_entity_type: string | null;
  related_entity_id: string | null;
  created_at: string;
}
